monkey/infection_monkey/i_puppet/target_host.py

```diff
@@ -35,17 +29,17 @@ def _filter_ports_by_status(self, status: PortStatus) -> Set[NetworkPort]:
 
 
 class TargetHostPorts(MutableInfectionMonkeyBaseModel):
-    class Config(MutableInfectionMonkeyModelConfig):
-        arbitrary_types_allowed = True
+    model_config = ConfigDict(arbitrary_types_allowed=True)
 
     tcp_ports: PortScanDataDict = Field(default_factory=PortScanDataDict)
     udp_ports: PortScanDataDict = Field(default_factory=PortScanDataDict)
 
+    @field_serializer("tcp_ports", "udp_ports", when_used="json")
+    def dump_ports(self, v):
+        return dict(v)
```

Comment:
Why do we need this function?

Reply:
Previously we had json_encoder, now we have serializer. Without this when serializing TargetHost it won't know to what to serialize the ports.

![image](https://github.com/guardicore/monkey/assets/15820737/0cc026b1-64c2-42d6-b771-32cf41cbbc55)